presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,703 @@
+            throw new PrestoException(NOT_SUPPORTED, "Renaming partition columns is not supported");
+        }
+
+        ImmutableList.Builder<Column> newDataColumns = ImmutableList.builder();
+        for (Column column : oldTable.getDataColumns()) {
+            if (column.getName().equals(oldColumnName)) {
+                newDataColumns.add(new Column(newColumnName, column.getType(), column.getComment()));
+            }
+            else {
+                newDataColumns.add(column);
+            }
+        }
+
+        Table newTable = Table.builder(oldTable)
+                .setDataColumns(newDataColumns.build())
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    @Override
+    public void dropColumn(String databaseName, String tableName, String columnName)
+    {
+        verifyCanDropColumn(this, databaseName, tableName, columnName);
+        Table oldTable = getTableOrElseThrow(databaseName, tableName);
+
+        if (!oldTable.getColumn(columnName).isPresent()) {
+            SchemaTableName name = new SchemaTableName(databaseName, tableName);
+            throw new ColumnNotFoundException(name, columnName);
+        }
+
+        ImmutableList.Builder<Column> newDataColumns = ImmutableList.builder();
+        for (Column fieldSchema : oldTable.getDataColumns()) {
+            if (!fieldSchema.getName().equals(columnName)) {
+                newDataColumns.add(fieldSchema);
+            }
+        }
+
+        Table newTable = Table.builder(oldTable)
+                .setDataColumns(newDataColumns.build())
+                .build();
+        replaceTable(databaseName, tableName, newTable, null);
+    }
+
+    @Override
+    public Optional<Partition> getPartition(String databaseName, String tableName, List<String> partitionValues)
+    {
+        try {
+            GetPartitionResult result = glueClient.getPartition(new GetPartitionRequest()
+                    .withDatabaseName(databaseName)
+                    .withTableName(tableName)
+                    .withPartitionValues(partitionValues));
+            return Optional.of(GlueToPrestoConverter.convertPartition(result.getPartition()));
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<List<String>> getPartitionNames(String databaseName, String tableName)
+    {
+        Table table = getTableOrElseThrow(databaseName, tableName);
+        List<Partition> partitions = getPartitions(databaseName, tableName, WILDCARD_EXPRESSION);
+        return Optional.of(buildPartitionNames(table.getPartitionColumns(), partitions));
+    }
+
+    /**
+     * @param parts List of partition values to filter on which can be full or partial.
```

Comment:
Nit: make this valid Javadoc. Put `@param` after the description and wrap the text in `<pre>` tags:
```java
/**
 * <pre>
 * ex: partition keys = ['a', 'b', 'c']  
 * parts could be:
 * ['1','2','3'] or
 * ['', '2', '']
 * </pre>
 * @param parts List of partition values to filter on which can be full or partial
 * @return a list of partition names
 */

Reply:
Will fix javadocs in this class